dash/App/Chains/Chain/ChainExpanded/index.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react'
+import link from '../../../../../resources/link'
+
+import Connection from '../Connection'
+
+import { SubmitChainButton, ChainHeader, EditChainColor, EditChainName, EditChainSymbol, EditChainId, EditTestnet, EditChainExplorer, ChainFooter } from '../Components'
+
+export default (props) => {
+  // props
+  const { id, name, type, explorer, symbol, isTestnet, filter, on, connection, primaryColor, icon, price } = props
+  const chain = { id, type, name, isTestnet, symbol, explorer, primaryColor }
+
+  // state
+  const [currentColor, setPrimaryColor] = useState(primaryColor)
+  const [currentName, setName] = useState(name)
+  const [currentSymbol, setSymbol] = useState(symbol)
+  const [currentExplorer, setExplorer] = useState(explorer)
+  const [currentTestnet, setTestnet] = useState(isTestnet)
+
+  useEffect(() => {
+    const updatedChain = { id, type, name: currentName, primaryColor: currentColor, isTestnet: currentTestnet, symbol: currentSymbol, explorer: currentExplorer }
+    link.send('tray:action', 'updateNetwork', chain, updatedChain)
+  }, [currentColor, currentName, currentSymbol, currentExplorer, currentTestnet])
+
+  const isMainnet = id === 1
+
+  return (
+    <div key={'expandedChain'} className='network cardShow'>
+      <ChainHeader 
+        type={type}
+        id={id}
+        primaryColor={currentColor}
+        icon={icon}
+        name={currentName}
+        on={on}
+        showToggle={true}
+      />
+      <EditChainColor 
+        currentColor={primaryColor} 
+        onChange={color => {
+          setPrimaryColor(color)
+          link.send('tray:action', 'setChainColor', chain.id, color)
```

Comment:
this can be another effect. after you started using `useEffect` I took a stroll through the React docs and it seems pretty common to have multiple effects that are very specific and targeted